lib/client/getPurchase.js

```diff
@@ -0,0 +1,26 @@
+const _validate = require('../_validate')
+
+const CONTEXT = 'getPurchaseDetails'
+
+module.exports = async (xhr, contextId, params = {}) => {
+  _validate({ xhr, contextId }, CONTEXT)
+
+  const qs = Object.keys(params)
+    .map(field => `${encodeURI(field)}=${encodeURI(params[field])}`)
+    .join('&')
```

Comment:
I think I've seen this several times ... can we move it inside xhr or as a helper function to map from object to query param ?